target/linux/ath79/dts/qca9558_belkin_f9j1108-v2.dts

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca955x.dtsi"
+
+/ {
+	model = "Belkin F9J1108v2 (AC1750 DB Wi-Fi)";
+	compatible = "belkin,f9j1108-v2", "qca,qca9558";
+
+	aliases {
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+		label-mac-device = &eth1;
+	};
+	
+	leds {
+		compatible = "gpio-leds";
+		
+		usb {
+			label = "f9j1108-v2:green:usb2";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&hub_port0>;
+			linux,default-trigger = "usbport";
+		};
+		
+		wps-amber {
+			label = "f9j1108-v2:amber:wps";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+		
+		wps-blue {
+			label = "f9j1108-v2:blue:wps";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+		};
+
+		led_system: system {
+			label = "f9j1108-v2:blue:system";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+	};
+	
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		gpio_usb_power {
+			gpio-export,name = "f9j1108-v2:usb2:power";
+			gpio-export,output = <1>;
+			gpios = <&gpio 21 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+		
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+	
+	virtual_flash {
+		compatible = "mtd-concat";
+
+		devices = <&fwpart1 &fwpart2>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				compatible = "openwrt,okli";
+				label = "firmware";
+				reg = <0x0 0x0>;
+			};
+		};
+	};
+};
+
+&pcie1 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&usb_phy0 {
+	status = "okay";
+};
+
+&usb0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port0: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "winbond,w25q128", "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <80000000>;
+		m25p,fast-read;
+
+		mtdparts: partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			
+			uboot: partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+			};
+			
+			fwpart1: partition@50000 {
+				label = "fwpart1";
+				reg = <0x050000 0xe20000>;
+			};
+			partition@e70000 {
+				label = "loader";
+				reg = <0xe70000 0x10000>;
+				read-only;
+			};
+			fwpart2: partition@e80000 {
+				label = "fwpart2";
+				reg = <0xe80000 0x150000>;
```

Comment:
What about 0xfd0000 to 0xff0000?

Reply:
Added 20000 to fwpart2, thanks!

Reply:
But was this just a mistake or is there something else supposed to be in that gap?

Reply:
It was a mistake, the original uImage partition goes right up to the ART partition:
0x000000050000-0x000000e70000 : "rootfs"
0x000000e70000-0x000000ff0000 : "uImage"
0x000000ff0000-0x000001000000 : "ART"

There are some factory configuration information seemingly contained inside rootfs and uImage, but I don't think we need to keep those safe, do we ?

Creating 1 MTD partitions on "rootfs":
0x000000e00000-0x000000e10000 : "EDINVRAMBKP"
Creating 1 MTD partitions on "rootfs":
0x000000e10000-0x000000e20000 : "EDIENVRAMBKP"
Creating 1 MTD partitions on "uImage":
0x000000160000-0x000000170000 : "EDINVRAM"
Creating 1 MTD partitions on "uImage":
0x000000170000-0x000000180000 : "EDIENVRAM"

Reply:
Inspecting the factory flash dump:
I can see key/value config data at offset 0xE50000 (which relates to the EDINVRAMBKP)  until 0xE528B3 followed by 00 until 0xE70000 where the kernel starts.
More key/value config data can be found at offset 0xFD0000 (which relates to EDINVRAM) until 0xFD28E8 followed by 00 until the end.
EDIENVRAMBKP and EDIENVRAM are both filled with 00.